editoast/src/views/timetable/similar_schedules.rs

```diff
@@ -0,0 +1,145 @@
+use axum::Extension;
+use axum::Json;
+use axum::extract::State;
+use chrono::DateTime;
+use chrono::Utc;
+use editoast_authz::Role;
+use serde::Deserialize;
+use serde::Serialize;
+use utoipa::ToSchema;
+
+use crate::error::Result;
+
+use super::AppState;
+use super::AuthenticationExt;
+use super::AuthorizationError;
+
+editoast_common::schemas! {
+    Waypoint,
+    WaypointResponse,
+}
+
+crate::routes! {
+    "/similar_schedules" => {
+        similar_schedules,
+    },
+}
+
+#[derive(Debug, Deserialize, ToSchema)]
+#[allow(dead_code)]
+struct RollingStockCharacteristics {
+    name: String,
+    speed_limit_tag: Option<String>,
+}
+
+#[derive(Clone, Deserialize, ToSchema)]
+#[cfg_attr(test, derive(PartialEq))]
+#[schema(as = SimilarScheduleWaypoint)]
+struct Waypoint {
+    ci: i64,
+    ch: String,
+    stop: bool,
+}
+
+impl std::fmt::Debug for Waypoint {
```

Comment:
nitpick: It looks more like a `Display` implementation than a `Debug`, isn't it?

Reply:
This PR is an extract of https://github.com/OpenRailAssociation/osrd/pull/11460 where I implemented Debug like this to display structures containing waypoint lists using `dbg!` a bit more concisely.

PS: I'll clean and split the PR before submitting it to review, ofc ;)

Reply:
Not invalidating the comment though, just explains the use case for Debug

Reply:
Not particularly attached to it, seems more semantically correct, but if the most probably use case is to use `dbg!()`, then let's keep it like this. I'm resolving.